mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -144,4 +146,31 @@ def GENX_BarrierOp : GENX_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
+def ShflKindXor : I32EnumAttrCase<"XOR", 0>;
+def ShflKindUp   : I32EnumAttrCase<"UP", 1>;
+def ShflKindDown : I32EnumAttrCase<"DOWN", 2>;
+
+/// Enum attribute of the different shuffle kinds.
+def ShflKind : I32EnumAttr<"ShflKind", "GENX shuffle kind",
+  [ShflKindXor, ShflKindUp, ShflKindDown]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def ShflKindAttr : EnumAttr<GENX_Dialect, ShflKind, "shfl_kind">;
+
+def GENX_SubGroupShuffleOp : GENX_Op<"sub_group_shuffle", [
+      TypesMatchWith<"result and value have the same type",
+                     "res", "value", "$_self">]>,
+  Results<(outs LLVM_Type:$res)>,
```

Comment:
LLVM_Type can be a struct, a ptr type, etc... Can we be more restrictive and only allow integer and float types ?